static/js/Misc.jsx

```diff
@@ -1103,96 +1104,114 @@ function useHiddenButtons() {
+      // then sort the refs and take only first 30 sources because admins don't want to reorder hundreds of sources
+      refs = refs.sort((a, b) => refSort('relevance', [a.ref, a], [b.ref, b])).slice(0, 30);
+      return <ReorderEditor close={toggleReorderCategory}
+                                        postURL={url}
+                                        type={'sources'}
+                                        origItems={refs}
+                                        redirect={`/topics/${data.slug}`}/>;
+  }
 
   const adminClasses = classNames({adminButtons: 1, hiddenButtons});
+  let wrapper = "";
   let adminButtonsSpan = null;
   if (Sefaria.is_moderator && editCategory) {
-    if (data.length === 0) {  // at /texts or /topics
-      const url = type === 'topics' ? `/api/topic/reorder` : `/api/category?reorder=1`;
-      const redirect = type === 'topics' ? '/topics' : '/texts';
-      const origItems = type === 'topics' ? Sefaria.topic_toc : Sefaria.toc;
-      adminButtonsSpan = <ReorderEditor close={toggleEditCategory} type={type} origItems={origItems}
-                                        postURL={url} redirect={redirect}/>;
-    } else if (type === "sources") {
-      const [topicSlug, refData] = data;
-      adminButtonsSpan = <SourceEditor topic={topicSlug} origData={refData} close={toggleEditCategory}/>;
-    } else if (type === "books") {
-      let tocObject = Sefaria.tocObjectByCategories(data);
-      const origDesc = {en: tocObject.enDesc, he: tocObject.heDesc};
-      const origCategoryDesc = {en: tocObject.enShortDesc, he: tocObject.heShortDesc};
-      const origData = {
-        origEn: tocObject.category,
-        origHe: tocObject.heCategory,
-        origDesc,
-        origCategoryDesc,
-        isPrimary: tocObject.isPrimary
-      };
-      adminButtonsSpan = <CategoryEditor origData={origData} close={toggleEditCategory} origPath={data.slice(0, -1)}/>;
-    } else if (type === "topics") {
-      const initCatSlug = TopicToCategorySlug(data);
-      const origData = {
-        origSlug: data.slug, origCategorySlug: initCatSlug,
-        origEn: data.primaryTitle.en, origHe: data.primaryTitle.he || ""
-      };
-      origData.origDesc = data.description || {"en": "", "he": ""};
-      origData.origCategoryDesc = data.categoryDescription || {"en": "", "he": ""};
-      const origWasCat = "displays-above" in data?.links;
-      adminButtonsSpan = <TopicEditor origData={origData}
-                                      origWasCat={origWasCat}
-                                      onCreateSuccess={(slug) => window.location.href = "/topics/" + slug}
-                                      close={toggleEditCategory}/>;
-    }
-  } else if (Sefaria.is_moderator && addSource && type === "topics") {
-    adminButtonsSpan = <SourceEditor topic={data.slug} close={toggleAddSource}/>;
+      adminButtonsSpan = getButtonForEditingCategory();
```

Comment:
from what I see, everything here is under a condition of `Sefaria.is_moderator` meaning we can have that condition at the top, and all the sub conditions will determine the adminButtonSpan. As the remaining code would be a single condition checking a variable name, a cleaner way to do it would be using a "switch case" condition https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch

Now the initial if statement checks it the user is an admin and will also have a "else" condition controlling for a non admin user (if that is necessary, perhaps not) and the "switch case" condition choses the component.